### PR TITLE
Add padding to DropdownButton

### LINKED ISSUE #115793: add padding to DropdownButton
## Use case

Inside the implementation of DropdownButton is an Inkwell. This Inkwell has a hitbox that does not cover the entire area of the button. A user can click on the button but the click is not handled.

## Proposal

Example - when implementing a custom dropdown that looks like this:
Screenshot from 2022-11-21 16-20-15

The clickable hitbox looks like this:
Screenshot from 2022-11-21 16-19-41

The desired hitbox should look like this:
Screenshot from 2022-11-21 16-20-02

To accomplish this, add Padding to the Inkwell.

### PR DESCRIPTION
**Pull request by davidskelly**
Add a padding param to DropdownButton. The current clickable hitbox looks like this when using external padding:

Screenshot from 2022-11-21 16-19-41

This allows the hitbox to look like this:

Screenshot from 2022-11-21 16-20-02

Fixes #115793

*If you had to change anything in the [flutter/tests] repo, include a link to the migration guide as per the [breaking change policy].*

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [X] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Review by Piinks — COMMENTED** [review]
Hey @davidskelly, welcome! Thanks for contributing! 
This looks like a bug that should work automatically instead of having the user fix it through a `padding` parameter. Can you remove the padding parameter and just adding the padding in the widget so the space is correct?

**Review by Piinks — COMMENTED** [review]
Thanks @davidskelly for updating, and sorry this one fell off my radar! Can you take a look at the failing analyzer check here?

**Comment by davidskelly** [comment]
@Piinks  - updated test!

**Comment by Piinks** [comment]
> This allows the hitbox to look like this:
> 
> Screenshot from 2022-11-21 16-20-02

I am still not sure why this cannot be handled automatically in the widget. 

What happens if the user provides padding that exceeds the bounds of the box?

File: packages/flutter/lib/src/material/dropdown.dart [hunk 1]
```diff
@@ -1128,6 +1130,11 @@ class DropdownButton<T> extends StatefulWidget {
   /// instead.
   final Color? dropdownColor;
 
+  /// Padding of this widget.
```
Comment (Piinks) [inline packages/flutter/lib/src/material/dropdown.dart:1133]: What part of the widget?

How is this different from the user just wrapping the `items` in padding?

**Comment by davidskelly** [comment]
> I am still not sure why this cannot be handled automatically in the widget.

You are suggesting that internally extra padding is just added - say a padding of 10 - instead of being passed as params?

The dropdown without my custom outline looks like this:
Screenshot from 2023-01-05 16-40-17

If internal padding is added, it would look something like this, which I don't think would be what most developers would want:
Screenshot from 2023-01-05 16-41-53


> What happens if the user provides padding that exceeds the bounds of the box?

The padding would grow the bounds of the box- the widget would just get bigger.

File: packages/flutter/lib/src/material/dropdown.dart (hunk 1 above)
Comment (bdlukaa) [reply packages/flutter/lib/src/material/dropdown.dart:1133]: > How is this different from the user just wrapping the items in padding?

The popup menu already adds a padding to the box. If `items` are wrapped in a `Padding`, it produces a weird effect in the popup (yet, looking good on the button). On the other hand, if no `Padding` is added, the popup looks good, but the button does not.

I believe that adding the same padding added to the popup menu to the button is a good fix for the issue, which is what this PR does

**Comment by davidskelly** [comment]
@Piinks is there anything else you need from me here?

File: packages/flutter/lib/src/material/dropdown.dart (hunk 1 above)
Comment (Piinks) [reply packages/flutter/lib/src/material/dropdown.dart:1133]: This documentation is still lacking some more context from my first comment here. :)

Comment (davidskelly) [reply packages/flutter/lib/src/material/dropdown.dart]: Pushed a change to update the documentation.

Comment (Piinks) [reply packages/flutter/lib/src/material/dropdown.dart]: Thank you! It isn't actually the padding around the button though, is it? It is inside of the button.
The docs should still also include an explanation of how this is different from just wrapping the button, or the content of the button, in a Padding widget. As well as, how does the padding affect the overall size of the button? 

All of the things we have been discussing in this PR make for helpful documentation for developers. :)

**Review by Piinks — APPROVED** [review]
LGTM, thank you. This will need a second reviewer, I will seek one out.

**Comment by esouthren** [comment]
The change LGTM, but I think it's worth mentioning for future readers that the custom Material 3 widget `DropDownMenu`  was created (https://github.com/flutter/flutter/pull/116088) which solves problems like this, by adding `DropdownMenuThemeData` which allows a custom `inputDecorationTheme`. 

I'd highly recommend using that widget over `DropdownButton` - it also works successfully in Material 2.

**Review by esouthren — APPROVED** [review]

**Pull request merged**